app/models/round.rb

```diff
@@ -121,6 +121,42 @@ def name
     Round.name_from_attributes(event, round_type)
   end
 
+  def registrations
+    if number == 1
+      Registration.joins(:registration_competition_events)
+                  .where(
+                    competition_id: competition_event.competition_id,
+                    competing_status: 'accepted',
+                    registration_competition_events: { competition_event_id: competition_event_id }
+                  ).includes([:user])
+    else
+      previous_round = Round.joins(:competition_event).find_by(competition_event: { competition_id: competition_event.competition_id, event_id: event.id }, number: number - 1)
+      previous_round.live_results.where(advancing: true).includes(:registration).map(&:registration)
+    end
+  end
+
+  def registrations_with_wcif_id
```

Comment:
Why do you need both `registrations` and also `registrations_with_wcif_id`?

Reply:
Also, we should probably take this horrendous query as an incentive to think about a better way for coming up with registration IDs.

Reply:
registrations -> loading data for front office, registrations_with_wcif_id -> loading data for back office